source/Irrlicht/COpenGLTexture.cpp

```diff
@@ -341,6 +341,12 @@ void COpenGLTexture::getOpenGLFormatAndParametersFromColorFormat(const asset::E_
 			colorformat = GL_RG;
 			type = GL_FLOAT;
 		}
+			case asset::EF_R8G8B8A8_SRGB:

```

Comment:
@qbasa12 there are more SRGB formats https://github.com/Crisspl/IrrlichtBAW/blob/master/include/irr/asset/EFormat.h, please provide cases for all.l

P.S. @Crisspl why are there EF_R8_SRGB, EF_R8G8_SRGB.... I'm 100% sure that OpenGL doesn't support that, is that even a thing in Vulkan?

Reply:
>there are more SRGB formats

The rest is taken care of or not supported in GL

>why are there EF_R8_SRGB, EF_R8G8_SRGB.... I'm 100% sure that OpenGL doesn't support that, is that even a thing in Vulkan?

All we have is slightly renamed VkFormat. Ofc it doesn't mean they are supported in GL

Reply:
I'm surprised those things are supported in vulkan XD.

However those switch statements are still missing the cases for `EF_R8G8B8_SRGB`, `EF_B8G8R8_SRGB`, the BCn SRGB formats, ETC and ASTC

Reply:
ETC and ASTC SRGB are all there, but if I haven't overlooked a few of BCn are missing (BC1-BC3 I think) so please add also them, thanks

Reply:
@qbasa12 any questions?